bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/handler/BaseModuleHandlerFactory.java

```diff
@@ -35,7 +35,7 @@
     @NonNullByDefault({})
     protected BundleContext bundleContext;
 
-    public void activate(BundleContext bundleContext) {
+    public void activate(@Nullable BundleContext bundleContext) {
```

Comment:
You mark the bundle context as nullable, but throw a IAE if it is null?
Then the argument is not nullable (IMHO).
Nullable should be used if null is accepted and valid to use.

Reply:
Actually the "protected BundleContext bundleContext" is used only in "activate" method here in this class, this fact is an unambiguous sign that this field should not be placed here, it is an implementation detail and every factory has to decide if it needs such